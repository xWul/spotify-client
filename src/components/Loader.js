import React from 'react'
import styled, { keyframes } from 'styled-components'

const Showbox = styled.div`
  background-color: rgba(0, 0, 0, .6);
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 5%;
  display: flex;
  justify-content: center;
  align-content: center;
  align-items: center;
  z-index: 9999;
`

const Loader = styled.div`
  position: relative;
  margin: 0 auto;
  width: 100px;
  &:before {
    content: '';
    display: block;
    padding-top: 100%;
  }
`

const spin = keyframes`
  to { transform: rotate(360deg);}
`
const Spinner = styled.div`
  border: 8px solid rgba(200, 200, 200, 0.1);
  border-left-color: #6290C8;
  border-radius: 50%;
  width: 100px;
  height: 100px;
  animation: ${spin} 1.2s linear infinite;
`

const ModalLoader = () => {
  return (
    <Showbox>
      <Loader>
        <Spinner />
      </Loader>
    </Showbox>
  )
}

export default ModalLoader