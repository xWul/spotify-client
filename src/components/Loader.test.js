import React from 'react'
import renderer from 'react-test-renderer'
import Loader from './Loader'

test('Loader renderer without crashing', () => {
  const component = renderer.create(
    <Loader />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
