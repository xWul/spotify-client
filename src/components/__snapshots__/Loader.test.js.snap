// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renderer without crashing 1`] = `
<div
  className="sc-bdVaJa dUnqUM"
>
  <div
    className="sc-bwzfXH grFctl"
  >
    <div
      className="sc-htpNat kVgGua"
    />
  </div>
</div>
`;
